kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java

```diff
@@ -21,14 +23,19 @@ public static String sanitizeName(String name) {
   // Note: a.b and a/b will have the same value after sanitization which will cause Duplicate key
   // Exception.
   public static Map<String, Object> replaceInvalidKeys(Map<String, Object> map) {
-    return map.entrySet().stream().collect(Collectors.toMap(
-        (entry) -> sanitizeName(entry.getKey()),
-        (entry) -> {
-          if (entry.getValue() instanceof Map) {
-            return replaceInvalidKeys((Map<String, Object>) entry.getValue());
+    try {
+
+      return map.entrySet().stream().collect(Collectors.toMap(
+          (entry) -> sanitizeName(entry.getKey()),
+          (entry) -> {
+            if (entry.getValue() instanceof Map) {
+              return replaceInvalidKeys((Map<String, Object>) entry.getValue());
+            }
+            return entry.getValue();
           }
-          return entry.getValue();
-        }
-    ));
+      ));
+    } catch (NullPointerException npe) {
+      throw new ConnectException("Records cannot have null keys");
+    }
```

Comment:
Instead of catching an unchecked exception, can you check null explicitly and throw an exception?